Code/Framework/AzQtComponents/AzQtComponents/Components/FancyDockingGhostWidget.cpp

```diff
@@ -76,11 +77,47 @@ namespace AzQtComponents
             window->setScreen(screen);
         }
 
-        setGeometry(targetRect);
+        QPoint midPoint = targetRect.topLeft() + QPoint(targetRect.width() / 2, targetRect.height() / 2);
+        QScreen* pointScreen = QApplication::screenAt(midPoint);
+        QRect rect(targetRect);
+
+        // On environment with multiple screens with different scaling settings, the screen coordinate system may have gaps
+        // due to the screen real estate shrinking according to the scale. When that happens, if a widget is moved into the gap
+        // it will resize and translate with undefined behavior, causing a lot of jitter and flashes.
+        // To prevent this, whenever the widget would end up outside screen boundaries, we resize the widget to be twice its
+        // original size so that the center of the widget is back inside the screen boundaries, and set the ghost widget
```

Comment:
If we see further issues with this we may want to look into a "one full-screen widget per screen" approach where we can paint appropriately, but I can't actually find an edge case with this approach so I'm OK with this.

Reply:
There are still some edge cases that will give some weird results, but much less than before.
The "one full-screen widget per screen" approach may still be useful for other cases, but those fell more into the realm of "needing to paint an element for a widget that is owned by screen 1 on screen 2" which are out of scope for this Epic. We actually do create full screen widgets for each screens in the Style Manager setup, so I guess they may already be used for something?